### PR TITLE
SAI-4630: now that it's possible to configure external tlogDir, we must delete them

### PR DESCRIPTION
**Pull request by magibney**
see [SOLR-16962](https://issues.apache.org/jira/browse/SOLR-16962)/[SOLR-6537](https://issues.apache.org/jira/browse/SOLR-6537)

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/core/SolrCore.java [hunk 1]
```diff
@@ -3356,6 +3382,23 @@ public static void deleteUnloadedCore(
       }
     }
     if (deleteInstanceDir) {
+      String ulogDir = cd.getUlogDir();
+      if (ulogDir != null) {
+        Path ulogDirPath = Path.of(ulogDir);
+        if (!ulogDirPath.startsWith(cd.getInstanceDir())
```
Comment (hiteshk25) [inline solr/core/src/java/org/apache/solr/core/SolrCore.java:3388]: Looks like above has same logic? maybe put in one function?

**Comment by hiteshk25** [comment]
@magibney this PR is still valid?

**Comment by magibney** [comment]
I'm going to bring it in line with upstream, which has changed a fair amount.

**Comment by magibney** [comment]
Closing this PR as redundant to #183

**Pull request closed without merging**